### ISSUE TITLE
train_shapes_gt_layout.py gets only 66% on SHAPES test set

### ISSUE THREAD

**Issue by rizar** [ronghanghu/n2nmn#7]
I am trying to reproduce SHAPES results from the paper by carefully following instructions from `README.md`. I made sure to install the recommended versions of TensorFlow and TensorFold in a fresh conda environment. Then I ran 

```
PYTHONPATH=$HOME/Dist/n2nmn python $HOME/Dist/n2nmn/exp_shapes/train_shapes_gt_layout.py
```

and saw the following learning curve:

![image](https://user-images.githubusercontent.com/654434/37614283-ba3463cc-2b80-11e8-9829-69ed25a8c383.png)

Then I tested the model using 

```
PYTHONPATH=$HOME/Dist/n2nmn python $HOME/Dist/n2nmn/exp_shapes/eval_shapes.py --exp_name shapes_gt_layout --snapshot_name 00040000 --test_split test
```

and got 

```
answer accuracy = 0.6708984375 on test
layout accuracy = 1.0 on test
layout validity = 1.0 on test
```

As far as I understand from the paper, the result of this experiment should be 100% accuracy on the test set, not 67%. Can you please help?

**Comment by ronghanghu** [comment]
Hi,
Usually the accuracy should quickly go up to ~100% in the learning curve. I remember that the training code sometimes got stuck in a poor local minima. Could you run it one or a few more times and see if it converges better?

**Comment by rizar** [comment]
Fair enough, 2 out of 5 runs converged in less than 1000 iterations. Thank again for your quick response!

![image](https://user-images.githubusercontent.com/654434/37616767-58e2fc08-2b87-11e8-9d84-3335c31a732b.png)
